hw/ip/i2c/doc/_index.md

```diff
@@ -263,6 +265,109 @@ However, all transitions which are dependent on formatting flags are shown expli
 
 ![](I2C_state_diagram.svg)
 
+# Programmers guide
```

Comment:
Programmers Guide created to hold examples of timing parameter generation.   C examples to follow in a later pull request.